src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -67981,11 +68002,45 @@ T M1<T>(T t = default)
     T M6<T>(T? t = default(T)) where T : struct // 7
         => t.Value; // 8
 }";
-            var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
+            var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue(), parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (5,12): warning CS8603: Possible null reference return.
+                //         => t; // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t").WithLocation(5, 12),
+                // (8,12): warning CS8603: Possible null reference return.
+                //         => t; // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t").WithLocation(8, 12),
+                // (10,19): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //     T M3<T>(T t = default) where T : class // 3
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "default").WithLocation(10, 19),
+                // (11,12): warning CS8603: Possible null reference return.
+                //         => t; // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t").WithLocation(11, 12),
+                // (14,12): warning CS8603: Possible null reference return.
+                //         => t; // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "t").WithLocation(14, 12),
+                // (17,12): warning CS8629: Nullable value type may be null.
+                //         => t.Value; // 6
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "t").WithLocation(17, 12),
+                // (19,16): error CS1770: A value of type 'T' cannot be used as default parameter for nullable parameter 't' because 'T' is not a simple type
+                //     T M6<T>(T? t = default(T)) where T : struct // 7
+                Diagnostic(ErrorCode.ERR_NoConversionForNubDefaultParam, "t").WithArguments("T", "t").WithLocation(19, 16),
+                // (20,12): warning CS8629: Nullable value type may be null.
+                //         => t.Value; // 8
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "t").WithLocation(20, 12)
+                );
+
+            comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics(
+                // (4,19): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //     T M1<T>(T t = default)
```

Comment:
nit: I think it would be easier to understand the expected diagnostics if they were renumbered based on whichever configuration produces the most diagnostics, so the other configurations can be judged based on what numbered diagnostics are missing.